<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QxrdIntegratorDialog</class>
 <widget class="QDockWidget" name="QxrdIntegratorDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>599</width>
    <height>607</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Integrate</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Oversampling Factor</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="3" colspan="2">
     <widget class="QSpinBox" name="m_OversampleFactor">
      <property name="toolTip">
       <string>Specify oversampling factor for angular integration
script property integrator.oversample</string>
      </property>
      <property name="specialValueText">
       <string>No oversampling</string>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>16</number>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_3">
      <property name="text">
       <string>Integrate vs</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="3" colspan="2">
     <widget class="QComboBox" name="m_IntegratorXUnits">
      <property name="toolTip">
       <string>Choose X units for integration
property integrator.integrationXUnits</string>
      </property>
      <item>
       <property name="text">
        <string>Integrate vs 2 Theta</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Integrate vs Q</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Integrate vs R</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Integration Step Size</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="2" column="3" colspan="2">
     <widget class="QxrdCenterStepSpinner" name="m_IntegratorStepSize">
      <property name="toolTip">
       <string>step size for integration
if ==  0, use fixed number of steps instead
script property integrator.integrationStep</string>
      </property>
      <property name="specialValueText">
       <string>Use fixed number of steps</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <widget class="QLabel" name="label_6">
      <property name="text">
       <string>Integration Number of Steps</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="3" column="3" colspan="2">
     <widget class="QSpinBox" name="m_IntegratorNSteps">
      <property name="toolTip">
       <string>Number of steps for integration
if == 0, use step size instead
script property integrator.integrationNSteps</string>
      </property>
      <property name="specialValueText">
       <string>Use fixed step size</string>
      </property>
      <property name="maximum">
       <number>999999</number>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Integration Minimum</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="4" column="3" colspan="2">
     <widget class="QDoubleSpinBox" name="m_IntegratorMinimum">
      <property name="toolTip">
       <string>Start value for integration
script property integrator.integrationMinimum</string>
      </property>
      <property name="decimals">
       <number>5</number>
      </property>
      <property name="maximum">
       <double>10000.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLabel" name="label_5">
      <property name="text">
       <string>Integration Maximum</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="5" column="3" colspan="2">
     <widget class="QDoubleSpinBox" name="m_IntegratorMaximum">
      <property name="toolTip">
       <string>Ending value for integration range
script property integrator.integrationMaximum</string>
      </property>
      <property name="decimals">
       <number>5</number>
      </property>
      <property name="maximum">
       <double>10000.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QPushButton" name="m_ClearGraphButton">
      <property name="toolTip">
       <string>Clear all curves from integrated data graph</string>
      </property>
      <property name="text">
       <string>Clear All Curves</string>
      </property>
     </widget>
    </item>
    <item row="7" column="4">
     <widget class="QPushButton" name="m_IntegrateOptionsButton">
      <property name="toolTip">
       <string>More intergration options</string>
      </property>
      <property name="text">
       <string>More Options...</string>
      </property>
     </widget>
    </item>
    <item row="8" column="0" colspan="2">
     <widget class="QCheckBox" name="m_EnableGeometry">
      <property name="toolTip">
       <string>Enable geometry corrections during integration
intensity *= pow(distance/m_DetectorDistance, 2)/cos(tth);</string>
      </property>
      <property name="text">
       <string>Enable Geometry Corrections </string>
      </property>
     </widget>
    </item>
    <item row="9" column="0">
     <widget class="QLabel" name="label_30">
      <property name="text">
       <string>User Geometry Kind</string>
      </property>
     </widget>
    </item>
    <item row="9" column="1" colspan="4">
     <widget class="QComboBox" name="m_EnableUserGeometry">
      <item>
       <property name="text">
        <string>None</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>User Defined Geometry</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="10" column="0">
     <widget class="QLabel" name="label_25">
      <property name="text">
       <string>User Geometry Func</string>
      </property>
     </widget>
    </item>
    <item row="10" column="3">
     <widget class="QLabel" name="label_27">
      <property name="text">
       <string>User Geometry Script</string>
      </property>
     </widget>
    </item>
    <item row="10" column="4">
     <widget class="QPushButton" name="m_UserGeometryEdit">
      <property name="text">
       <string>Edit...</string>
      </property>
     </widget>
    </item>
    <item row="11" column="0" colspan="2">
     <widget class="QCheckBox" name="m_EnablePolarization">
      <property name="toolTip">
       <string>Enable polarization correction during integration:
intensity /= (m_Polarization      *(1 - pow(sin(chi)*sin(tth/2),2)) +
                   (1 - m_Polarization)*(1 - pow(cos(chi)*sin(tth/2),2)));</string>
      </property>
      <property name="text">
       <string>Enable Polarization Corrections </string>
      </property>
     </widget>
    </item>
    <item row="11" column="3">
     <widget class="QLabel" name="label_24">
      <property name="text">
       <string>Polarization Factor</string>
      </property>
     </widget>
    </item>
    <item row="11" column="4">
     <widget class="QDoubleSpinBox" name="m_Polarization">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTip">
       <string>Polarization factor : [0..1]</string>
      </property>
      <property name="keyboardTracking">
       <bool>false</bool>
      </property>
      <property name="prefix">
       <string/>
      </property>
      <property name="decimals">
       <number>6</number>
      </property>
      <property name="maximum">
       <double>1.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
    </item>
    <item row="12" column="0" colspan="2">
     <widget class="QCheckBox" name="m_EnableAbsorption">
      <property name="toolTip">
       <string>Enable absorption correction during integration
intensity *= 
  exp((distance-m_DetectorDistance)/m_AttenuationLength);</string>
      </property>
      <property name="text">
       <string>Enable Absorption Corrections </string>
      </property>
     </widget>
    </item>
    <item row="12" column="3">
     <widget class="QLabel" name="label_23">
      <property name="text">
       <string>Attenuation Length</string>
      </property>
     </widget>
    </item>
    <item row="12" column="4">
     <widget class="QDoubleSpinBox" name="m_AttenuationLength">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTip">
       <string>Attenuation Length (mm)</string>
      </property>
      <property name="keyboardTracking">
       <bool>false</bool>
      </property>
      <property name="decimals">
       <number>6</number>
      </property>
      <property name="maximum">
       <double>1000000.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>1000.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="13" column="0">
     <widget class="QLabel" name="label_29">
      <property name="text">
       <string>User Absorption Kind</string>
      </property>
     </widget>
    </item>
    <item row="13" column="1" colspan="4">
     <widget class="QComboBox" name="m_EnableUserAbsorption">
      <item>
       <property name="text">
        <string>None</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>A function of pixel coordinates</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>A function of x,y displacement from beam center</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>A function of r,chi</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>A function of q,chi</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="14" column="0">
     <widget class="QLabel" name="label_26">
      <property name="text">
       <string>User Absorption Func</string>
      </property>
      <property name="wordWrap">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="14" column="3">
     <widget class="QLabel" name="label_28">
      <property name="text">
       <string>User Absorption Script</string>
      </property>
     </widget>
    </item>
    <item row="14" column="4">
     <widget class="QPushButton" name="m_UserAbsorptionEdit">
      <property name="text">
       <string>Edit...</string>
      </property>
     </widget>
    </item>
    <item row="15" column="3">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>248</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="7" column="3">
     <widget class="QPushButton" name="m_ClearSelectedGraphButton">
      <property name="toolTip">
       <string>Clear selected curves from integrated data graph
(Click legend entries to select curves)</string>
      </property>
      <property name="text">
       <string>Clear Selected Curves</string>
      </property>
     </widget>
    </item>
    <item row="7" column="1" colspan="2">
     <widget class="QPushButton" name="m_IntegrateButton">
      <property name="text">
       <string>Integrate</string>
      </property>
     </widget>
    </item>
    <item row="10" column="1" colspan="2">
     <widget class="QLineEdit" name="m_UserGeometryFunction">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="14" column="1" colspan="2">
     <widget class="QLineEdit" name="m_UserAbsorptionFunction">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QxrdCenterStepSpinner</class>
   <extends>QDoubleSpinBox</extends>
   <header>qxrdcenterstepspinner.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
